cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorManualRollingUpdatesTest.java

```diff
@@ -429,6 +430,156 @@ vertx, new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+
+        ClusterOperatorConfig config = ResourceUtils.dummyClusterOperatorConfig(featureGates);
+
+
+        MockZooKeeperReconciler zr = new MockZooKeeperReconciler(
+            new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+            vertx,
+            config,
+            supplier,
+            new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+            kafka,
+            VERSION_CHANGE,
+            null,
+            0,
+            null,
+            forceZookeeperError);
+
+        MockKafkaReconciler kr = new MockKafkaReconciler(
+            new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+            vertx,
+            config,
+            supplier,
+            new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+            kafka,
+            null,
+            kafkaCluster,
+            null,
+            null,
+            forceKafkaError);
+
+        MockKafkaAssemblyOperator kao = new MockKafkaAssemblyOperator(
+            vertx, new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+            CERT_MANAGER,
+            PASSWORD_GENERATOR,
+            supplier,
+            config,
+            zr,
+            kr);
+
+        Checkpoint async = context.checkpoint();
+        if (expectError) {
+            kao.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME))
+                .onComplete(context.failing(e -> context.verify(() -> {
+                    assertThat(e.getMessage(), containsString("Force failure"));
+                    async.flag();
+                })));
+        } else {
+            kao.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME))
+                .onComplete(context.succeeding(e -> context.verify(() -> {
+                    // Verify Zookeeper rolling updates
+                    assertThat(zr.maybeRollZooKeeperInvocations, is(1));
+                    assertThat(zr.zooPodNeedsRestart.apply(podWithName("my-cluster-zookeeper-0")), is(nullValue()));
+                    assertThat(zr.zooPodNeedsRestart.apply(podWithName("my-cluster-zookeeper-1")), is(List.of("manual rolling update annotation on a pod")));
+                    assertThat(zr.zooPodNeedsRestart.apply(podWithName("my-cluster-zookeeper-2")), is(List.of("manual rolling update annotation on a pod")));
+
+                    // Verify Kafka rolling updates
+                    assertThat(kr.maybeRollKafkaInvocations, is(1));
+                    assertThat(kr.kafkaNodesNeedRestart.size(), is(2));
+                    assertThat(kr.kafkaNodesNeedRestart, is(List.of("my-cluster-kafka-0", "my-cluster-kafka-1")));
+                    assertThat(kr.kafkaRestartReasons.apply(podWithName("anyName")), is(RestartReasons.of(RestartReason.MANUAL_ROLLING_UPDATE)));
+
+                    async.flag();
+                })));
+        }
+    }
```

Comment:
Maybe you could keep an empty line between the different tests?

Reply:
There is already an empty line between each new test methods.

Reply:
Yeah, but not between 582 and 583, or?